### PR TITLE
[JB][SUPP-352]

### PR DESCRIPTION
**Pull request by barnesjake**
- add cannot setup direct debit online page
- refactor the test data/json stuff

### CODE REVIEW

**Review by nubz — APPROVED** [review]
FE LGTM

File: test/testsupport/testdata/TdJsonBodies.scala [hunk 1]
```diff
@@ -56,1630 +67,305 @@ object TdJsonBodies {
       |        "backUrl" : "/set-up-a-payment-plan/test-only/bta-page?starting-page"
       |      }
       |    },
-      |    "sessionId": "IamATestSessionId",
-      |    "taxId": {
-      |      "value": "123/456"
-      |    }
+      |    "sessionId": "IamATestSessionId"
+      |    ${createJourneyInfoJson(journeyInfo, "")}
```
[the remark this answers is not in the document]
Comment (barnesjake) [reply test/testsupport/testdata/TdJsonBodies.scala:71]: very good point 😂 , I'll tinker

File: test/testsupport/testdata/StageInfo.scala [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package testsupport.testdata
+
+object StageInfo {
+  val computedTaxId: (String, String) = ("ComputedTaxId", "ComputedTaxId")
```
[the remark this answers is not in the document]
Comment (barnesjake) [reply test/testsupport/testdata/StageInfo.scala:20]: Will do

File: test/testsupport/testdata/TdJsonBodies.scala [hunk 3]
```diff
@@ -40,12 +41,14 @@ object TdJsonBodies {
     val detachedUrl: String = bta
   }
 
-  def afterDetermineTaxIdJourneyJson(): String =
-    """
+  def createJourneyJson(stageInfo: StageInfo, journeyInfo: List[String]): String = {
+    //drop the trailing comma
+    val jsonFormatted: String = journeyInfo.mkString(",").dropRight(1)
```
[the remark this answers is not in the document]
Comment (barnesjake) [reply test/testsupport/testdata/TdJsonBodies.scala]: lol sorry left that in while debugging something/tinkering, but just removed it and tests still pass, interesting how they did if I was using dropRight, surely it should have lost the closing " or something?

Updated to remove that though

[the remark this answers is not in the document]
Comment (AndyHWChung) [reply test/testsupport/testdata/TdJsonBodies.scala]: 🤔 that's odd - maybe there was a trailing space at the end or maybe the function was not being used in the tests? 🤷

[the remark this answers is not in the document]
Comment (barnesjake) [reply test/testsupport/testdata/TdJsonBodies.scala]: probably white space or something

**Pull request merged**